src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,152 @@
+                let mount_result = share_fs_mount
+                    .share_rootfs(ShareFsRootfsConfig {
+                        cid: cid.to_string(),
+                        source: extra_options.snapshot_dir.clone(),
+                        target: SNAPSHOT_DIR.to_string(),
+                        readonly: true,
+                        is_rafs: false,
+                    })
+                    .await
+                    .context("share nydus rootfs")?;
+                let mut options: Vec<String> = Vec::new();
+                options.push(
+                    "lowerdir=".to_string()
+                        + &do_get_guest_path(
+                            HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR,
+                            cid,
+                            false,
+                            true,
+                        ),
+                );
+                options.push(
+                    "workdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOT_DIR, "work").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push(
+                    "upperdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOT_DIR, "fs").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push("index=off".to_string());
+                Ok((
+                    Storage {
+                        driver: KATA_OVERLAY_DEV_TYPE.to_string(),
+                        source: TYPE_OVERLAY_FS.to_string(),
+                        fs_type: TYPE_OVERLAY_FS.to_string(),
+                        options,
+                        mount_point: mount_result.guest_path.clone(),
+                        ..Default::default()
+                    },
+                    mount_result.guest_path,
+                ))
+            }
+            NYDUS_ROOTFS_V6 => {
+                // TODO
+                // As some kernel patch is not support in the upstream community, we will delivery this feature later.
+                // see this issue (https://github.com/kata-containers/kata-containers/issues/5143)
+                let errstr: &str = "new_nydus_rootfs: nydus v6 unavailable currently";
+                error!(sl!(), "{}", errstr);
+                Err(anyhow!(errstr))
+            }
+            _ => {
+                let errstr: &str = "new_nydus_rootfs: invalid nydus rootfs type";
+                error!(sl!(), "{}", errstr);
+                Err(anyhow!(errstr))
+            }
+        }?;
+        Ok(NydusRootfs {
+            guest_path: rootfs_guest_path,
+            rootfs: rootfs_storage,
```

Comment:
Looks like guest_path is identical to rootfs.mount_point. Why do we need to save it separately?

Reply:
It's now really necessary cause only the `mount_point` in `storage` struct is used in agent. I will raise another issue to solve that